### PR TITLE
feat: add in app support widget

### PR DESCRIPTION
**Pull request by owlas**
### Description:
Adds Chatwoot - an open source widget to contact the Lightdash team. This is enabled by passing the `CHATWOOT_BASE_URL` and `CHATWOOT_TOKEN` variables

### Preview:

**Chat widget when configured to be visible but not open**
Screenshot 2021-12-13 at 17 07 03

**Chat widget when first opened**
Screenshot 2021-12-13 at 17 07 19

**Chat widget when sending a message and not logged in**
Asks for email
Screenshot 2021-12-13 at 17 07 34

**Chat widget when sending a message and user is logged in**

[todo]

### Scope of the changes (select all that apply):
- [x] Frontend  
- [x] Backend  
- [x] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD  

### Todo

* Allow visibility to be configurable
* Create a separate button (styled into the UI / navbar) to trigger chat to open (e.g. behave more like regular support)
* broken login

### CODE REVIEW

File: packages/frontend/src/providers/TrackingProvider.tsx [hunk 1]
```diff
@@ -210,6 +210,7 @@ export const TrackingProvider: FC<TrackingData> = ({
         ({ id, traits }: IdentifyData) => {
             if (mode && mode !== LightdashMode.DEMO) {
                 rudderAnalytics?.identify(id, traits, lightdashContext);
+                (window as any).$chatwoot.setUser(id, traits);
```
Comment (ZeRego) [inline packages/frontend/src/providers/TrackingProvider.tsx:213]:
```suggestion
                (window as any).$chatwoot?.setUser(id, traits);
```

File: packages/frontend/src/providers/AppProvider.tsx [hunk 2]
```diff
@@ -93,6 +94,45 @@ export const AppProvider: FC = ({ children }) => {
...
+            script.onload = () => {
+                (window as any).chatwootSDK.run({
+                    websiteToken: health.data.chatwoot.websiteToken,
+                    baseUrl: health.data.chatwoot.baseUrl,
+                });
+            };
+            setIsChatwootLoaded(true);
+            window.addEventListener('chatwoot:ready', identify);
+        }
+        identify();
```
Comment (ZeRego) [inline packages/frontend/src/providers/AppProvider.tsx:133]: For what scenario do we need to trigger the identify here ? 

- If the user loads a page and is unidentified, we load the chatwoot and later on they login so we identify the user in chatwoot via the TrackingProvider. 
- If the user loads a page and is identified, we load the chatwoot and once we get the 'chatwoot:ready' event we identify the user.

Comment (owlas) [reply packages/frontend/src/providers/AppProvider.tsx:133]: This is for the case where:
1. User is identified (but widget not loaded so it's a no-op)
2. Widget loads
3. User is identified again from `Chatwoot:ready` event (now widget has loaded and user is identified)

File: packages/frontend/src/providers/TrackingProvider.tsx (hunk 1 above)
Comment (owlas) [reply packages/frontend/src/providers/TrackingProvider.tsx:213]: Thanks, this line shouldn't be in the PR. I moved the logic to App Provider

**Comment by PriPatel** [comment]
I think _for now_ , we want to understand friction and therefore make it as easy as possible for users to get in touch. I don't think having the bubble in the **bottom-right on every page** is a problem! I've done a few tests and a bottom-right button doesn't seem to obstruct anything important. 

[Mockup](https://www.figma.com/proto/Mhuwki1Bj1akQIwPyr1gWw/Lightdash-product?node-id=598%3A31245&scaling=scale-down-width&page-id=593%3A30919&hide-ui=1
)  
[Figma file](https://www.figma.com/file/Mhuwki1Bj1akQIwPyr1gWw/Lightdash-product?node-id=593%3A30919) 

Changes to be made:
- The actual button. The standard Chatwoot button doesn't show a very clear speech bubble!
- Our footers have links to `Join the discussion!` with yet another 'speech bubble' icon. I propose we get rid of this and simply have A link to `Documentation`. This makes it real clear where to go when you want to chat to someone or you just want to read docs.

**Comment by owlas** [comment]
@PriPatel I've just realised that this might hover over our toasts (when they show "successfully saved!") - just trying to place the floating chat button between those. 

You also mentioned somewhere else in the app where the button placement doesn't work?

**Comment by owlas** [comment]
@PriPatel what do you think about using the blueprintjs chat icon? 
Screenshot 2021-12-17 at 10 35 58

**Comment by owlas** [comment]
@ZeRego I feel like there is some logic split between `OpenChatButton.tsx` and the `AppProvider` - what do you think?

**Comment by ZeRego** [comment]
> @ZeRego I feel like there is some logic split between `OpenChatButton.tsx` and the `AppProvider` - what do you think?

It would look cleaner if everything was inside the OpenChatButton component.

**Comment by PriPatel** [comment]
At @owlas - the icon and colour is fine - we can change this later.
Did you manage to make the change on the footer? (it uses exactly the same icon for `Join the conversation`!)

Lets hide it on :
`Explore pages`
`Project settings
`
`Dashboard pages
`

^ I know that sounds like quite a lot but now Im thinking it is going to be annoying. If we leave it on the onboarding pages for now, the user will know to ask for help on the homepage. I will redo the footer / top bar nav to make sure this works on every page at some point next week. I just need to set time aside for it as its a much bigger task!

**Comment by owlas** [comment]
Ok perfect! So...

> Did you manage to make the change on the footer? (it uses exactly the same icon for Join the conversation!

Will update the footer

> leave it on the onboarding pages for now 

Only have chat on onboarding and home page?

**Comment by PriPatel** [comment]
@owlas I think so! Just to be safe in the knowledge that we 're not obstructing anything/annoying users. I will find a solution for a more subtle, permanent place thats works holistically with the rest of our navigation! 
For now they can go to that 'homepage' when they want to chat. I think that's fine - I've seen this pattern before.

**Comment by owlas** [comment]
Updated.

**A page without chat and new footer icon**
explore-no-chat

**A page with chat minimised and new footer icon + text**
chat-closed

**A page with chat open and new footer icon + text**
chat-open

**Comment by owlas** [comment]
@zerego I've moved the identify logic out to the chat widget. We no longer identify just after the widget is loaded but now the chat is only on certain pages, I think we won't miss an identify event.

I've left the sdk loading logic in the AppProvider because I don't want to load it into the window multiple times.

File: packages/frontend/src/components/AboutFooter.tsx [hunk 3]
```diff
@@ -69,12 +69,12 @@ const AboutFooter: FC<{ minimal?: boolean; maxWidth?: number }> = ({
...
-                            href="https://community.lightdash.com/"
+                            href="https://docs.lightdash.com/"
                             target="_blank"
                             minimal
-                            icon="chat"
+                            icon="manual"
                         >
-                            {!minimal && 'Join the conversation!'}
+                            {!minimal && 'Documentation'}
                         </AnchorButton>
```
Comment (ZeRego) [inline packages/frontend/src/components/AboutFooter.tsx:78]: We should remove the `help` button that also redirects to the docs website

**Comment by owlas** [comment]
remove duplicate help button

**Review by ZeRego — APPROVED** [review]

**Pull request merged**